notebook/terminal/handlers.py

```diff
@@ -17,8 +23,53 @@ def get(self, term_name):
                    ws_path="terminals/websocket/%s" % term_name))
 
 class TermSocket(terminado.TermSocket, IPythonHandler):
+
     def set_default_headers(self):
         pass
+
+    def check_origin(self, origin):
```

Comment:
this seem to be identical to what is in notebook/base/handler:APIHandler.
Do you think we should move the method up the hierarchy ! class (to IPythonHandler) to avoid code duplication ?